.github/workflows/ci.yml

```diff
@@ -1087,6 +1087,104 @@ jobs:
           path: diagnostics*.gz
           retention-days: 1
 
+  # Runs tests on `tests/cli/fluvio_components`
+  fluvio_components:
+    name: Fluvio Components Test (${{ matrix.test }}) on ${{ matrix.os }} for ${{ matrix.rust-target }}
+    runs-on: ${{ matrix.os }}
+    env:
+      FLUVIO_BIN: "~/bin/fluvio"
+      SERVER_LOG: fluvio=info
+    needs:
+      - build_primary_binaries
+      - config
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+        rust-target:
+          - x86_64-unknown-linux-musl
+        test:
+          - multiple_partitions
+    steps:
+      - name: Checkout Source Code
+        uses: actions/checkout@v3
+
+      - name: Setup K3D
+        run: |
+          curl -s https://raw.githubusercontent.com/rancher/k3d/main/install.sh | TAG=${{ env.K3D_VERSION }} bash
+          ./k8-util/cluster/reset-k3d.sh
+
+      - name: Wait 15s for K3D Reset
+        run: sleep 15
+
+      # This will load into ~/.fluvio/bin which will be in PATH
+      - name: Install Stable CLI and start Fluvio cluster
+        run: |
+          curl -fsS https://hub.infinyon.cloud/install/install.sh | bash
+          echo "~/.fluvio/bin" >> $GITHUB_PATH
+
+      # Hacky: We do this to create the Fluvio Config, perhaps in the future
+      # we can introduce a `fluvio config init` command to do this instead
+      - name: Initialize Cluster Locally
+        run: fluvio cluster start
+
+      - name: Wait 10s for Cluster be Available
+        run: sleep 10
+
+      - name: Initialize Cluster Locally
+        run: fluvio cluster shutdown
+
+      # Download Artifacts from Development Build (This Commit)
+      - name: Download artifact - fluvio
+        uses: actions/download-artifact@v3
+        with:
+          name: fluvio-${{ matrix.rust-target }}
+          path: ~/bin
+
+      - name: Download artifact - fluvio-run
+        uses: actions/download-artifact@v3
+        with:
+          name: fluvio-run-${{ matrix.rust-target }}
+          path: ~/.fluvio/extensions
+
+      - name: Check Directory Contents
+        run: ls -a ~/bin
+
+      - name: Set up Fluvio Binaries
+        run: |
```

Comment:
why this is necessary? should use `fluvio cluster start --local``

Reply:
I want to use PRs Fluvio version for tests, so I am replacing the fluvio binary from the installation to use the one from the build artifacts